core/src/Revolution/Processors/Security/ResourceGroup/GetList.php

```diff
@@ -28,6 +31,48 @@ class GetList extends GetListProcessor
     public $languageTopics = ['access'];
     public $permission = 'resourcegroup_view';
 
+    /** @param boolean $isGridFilter Indicates the target of this list data is a filter field */
+    private $isGridFilter;
+
+    public function initialize()
+    {
+        $initialized = parent::initialize();
+        $this->isGridFilter = $this->getProperty('isGridFilter', false);
+        return $initialized;
+    }
+
+    /**
+     * {@inheritDoc}
+     * @param xPDOQuery $c
+     * @return xPDOQuery
+     */
+    public function prepareQueryBeforeCount(xPDOQuery $c)
+    {
+        /*
+            When this class is used to fetch data for a grid filter's store (combo),
+            limit results to only those resource groups present in the current grid.
+        */
+        if ($this->isGridFilter) {
+            if ($userGroup = $this->getProperty('usergroup', false)) {
+                $c->innerJoin(
+                    modAccessResourceGroup::class,
+                    'modAccessResourceGroup',
+                    [
+                        '`modAccessResourceGroup`.`target` = `modResourceGroup`.`id`',
+                        '`modAccessResourceGroup`.`principal` = ' . (int)$userGroup,
+                        '`modAccessResourceGroup`.`principal_class` = ' . $this->modx->quote(modUserGroup::class)
+                    ]
+                );
+            }
+            if ($policy = $this->getProperty('policy', false)) {
+                $c->where([
+                    '`modAccessResourceGroup`.`policy`' => (int)$policy
+                ]);
+            }
+        }
+        return $c;
```

Comment:
Have you considered to use an early return of `$c` instead of using a huge if statement block?